Add DataGrid docs (#1616)
mui/mui-toolpad: docs/data/toolpad/building-ui/datagrid-component.md

# DataGrid component

<p class="description">Build powerful and feature rich data tables</p>

<img src="/static/toolpad/docs/building-ui/datagrid.png" width="884px" alt="Custom component" />

## Customizing cell rendering

> By default you can render json data using DataGrid component and it will automatically recognize most of the data types. However sometimes it might not be enough and for that reason we provide the option to render custom components inside a datagrid.

Reviewer: The content seems fine to me! But for improving the grammar and punctuation we could run this by Sam maybe?

Reviewer 2: ```suggestion
By default, you can render JSON data using the Data Grid component and it will automatically recognize most of the data types. 
If this isn't sufficient, you can also render custom components inside the Data Grid.
```

### Creating custom component

Firstly, you will have to create a custom component which should be able to process `DataGrid` data:

- `value` prop will receive the value of the specific column
- `row` prop will receive the data for the entire row
- `field` prop will receive field configuration data

<img src="/static/toolpad/docs/building-ui/cell-component.png" width="621px" alt="Custom component" />

### Configuring columns

> 

Reviewer 3: Perhaps it could be interesting to have a section for each column type?

<img width="286" alt="Screenshot 2023-02-02 at 11 40 57" src="https://user-images.githubusercontent.com/2109932/216303009-2c63592b-1c42-47fb-ae61-eb1892f1059e.png">

# DataGrid component

<p class="description">Build powerful and feature rich data tables</p>

<img src="/static/toolpad/docs/building-ui/datagrid.png" width="884px" alt="Custom component" />

## Customizing cell rendering

By default, you can render JSON data using the Data Grid component and it will automatically recognize most of the data types. 
If this isn't sufficient, you can also render custom components inside the Data Grid.

### Creating custom component

Firstly, you will have to create a custom component which should be able to process `DataGrid` data:

- `value` prop will receive the value of the specific column
- `row` prop will receive the data for the entire row
- `field` prop will receive field configuration data

<img src="/static/toolpad/docs/building-ui/cell-component.png" width="621px" alt="Custom component" />

### Configuring columns

1. Select `DataGrid` component

> 2. Click on columns edit button

Reviewer: `Click on the "columns" button` maybe?

<img src="/static/toolpad/docs/building-ui/edit-columns.png" width="286px" alt="Edit columns" />

3. Choose the column

<img src="/static/toolpad/docs/building-ui/choose-column.png" width="314px" alt="Choose column" />

4. Change type to use codeComponent

<img src="/static/toolpad/docs/building-ui/choose-column-type.png" width="318px" alt="Choose column type" />

> 5. Choose custom component that you created

Reviewer: "Choose **the** custom component that you created"
